gtsam/linear/NoiseModel.h

```diff
@@ -718,15 +724,18 @@ namespace gtsam {
       virtual void WhitenSystem(Matrix& A1, Matrix& A2, Matrix& A3, Vector& b) const;
 
       virtual Vector unweightedWhiten(const Vector& v) const {
-        return noise_->unweightedWhiten(v);
+        return noise_->whiten(v);
```

Comment:
comments

Reply:
So many things going on. Why this change? I think the PR should be more focused on the new math write-up. Is this related?

Reply:
After changing the noise_ to gaussian, we can directly use the whiten method.

Here is how things are going on:
    In order to agree with the new math, we need to calculate the Mahalanobis distance for the robust noise model. Since mahalanobisDistance is only implmented for Gaussian, I changed the NoiseModel inside robust to Gaussian (previously was general noisemodel), which makes sense since it must be Gaussian.
    These changes follow the change of the NoiseModel to Gaussian.

Reply:
Yes, but unweightedWhiten was not used, right? why is this a thing in this PR? I think we should talk to @mikebosse whether unweightedWhiten is needed at all and have a PR just to remove them, if so?

Reply:
Sounds good, I'll change it back.